kit/commands/plugin.py

```diff
@@ -66,85 +68,147 @@ def get_plugin_type(plugin_path: Path) -> PluginType:
     raise TypeError("This program only supports tarball or zip files")
 
 
-def check_plugin_structure(plugin_path: Path, plugin_type: PluginType) -> str:
+def check_plugin_structure(
+    plugin_path: Path, plugin_type: PluginType
+) -> PluginSettingsList:
     """Check the minimum plugin structure (a directory with a plugin.py file)
     and return its name"""
-    plugin_name = ""
-    expected_file = "plugin.py"
-
-    if PluginType.DIR == plugin_type:
-        plugin_name = plugin_path.name
-        if not (plugin_path / expected_file).exists():
-            raise FileNotFoundError(
-                f"File '{expected_file}' not found in '{plugin_name}'"
-            )
-        return plugin_name
+    plugin_settings_list = []
+    exp_toml = "plugin.toml"
 
     try:
+        if PluginType.DIR == plugin_type:
+            exp_file = exp_toml
+            plugin_settings_dict = load_toml(plugin_path / exp_file)["plugin"]
+            exp_file = plugin_path / plugin_settings_dict["start"]
+            if not Path(exp_file).exists():
+                raise FileNotFoundError()
+            return [plugin_settings_dict]
+
         if PluginType.TAR == plugin_type:
             with tar_open(plugin_path) as f:
-                plugin_name = f.getmembers()[0].name
-                # getmember raises a KeyError if the file cannot be found
-                f.getmember(f"{plugin_name}/{expected_file}")
+                # get the list of plugin.toml files
+                exp_file = exp_toml
+                tar_toml_list = [
+                    item.name for item in f.getmembers() if exp_file in item.name
+                ]
+                if not tar_toml_list:
+                    raise FileNotFoundError()
+
+                # getmember and extractfile raise a KeyError if the file cannot be found
+                for toml_file in tar_toml_list:
+                    # Verify the toml file is in a directory
+                    exp_file = f"DIRECTORY/{exp_toml}"
+                    dir_name = toml_file.split("/")[0]
+                    f.getmember(dir_name)
+
+                    # Read the toml file and check that the file
+                    # defined in "start" is present
+                    with f.extractfile(toml_file) as file:  # type: ignore[union-attr]
```

Comment:
```suggestion
                    
                    # Read the toml file and check that the file
                    # defined in "start" is present
                    file = f.extractfile(toml_file)
                    if file == None:
                        raise TypeError(...)
                    with file: 
```
This fixes the mypy issue. union-attr is because extractfile returns an Optional[IO] type. In other words it is a union of a type IO and None. mypy checks that all the types of the union have the attribute being called.

Reply:
add appropriate msg to the raised error. dont just accept the suggestion :)